### PR TITLE
FIX: Parsing imagelayer inside a group tag in TMX file

### PR DESCRIPTION
**Pull request by TMalygin**
TmxMapLoader did not parse `imagelayer` tag when it is inside `group` tag.

The problem was that the code only passed through the descendants of the `map` tag.
The Solution is to change ` root.getChildByName("tileset")` to ` root.getChildrenByNameRecursively("tileset")`

Added the test for this situation

### CODE REVIEW

**Comment by BoBIsHere86** [comment]
Wow! I can't believe I didn't realize getChildrenByNameRecursively existed.
I had a manual fix for this here, PR https://github.com/libgdx/libgdx/pull/7075
Your solution is way cleaner, so I can close out mine out if this works.
Can you confirm this works on really messy and nested layer groups, something like this?
image

**Comment by TMalygin** [comment]
> Wow! I can't believe I didn't realize getChildrenByNameRecursively existed. I had a manual fix for this here, PR #7075 Your solution is way cleaner, so I can close out mine out if this works. Can you confirm this works on really messy and nested layer groups, something like this? image

Yes, I updated `TiledMapGroupLayerWithImagelayerTest`, so it uses the `test.tmx` same as from your comment.

The screenshot
Screenshot from 2023-05-02 20-01-13
[test.tmx](https://github.com/libgdx/libgdx/pull/7135/files#diff-6570cba33c9fea6beca8bd1e8dea81554019ebf4a490fbac74e4c4f83e3fc8f1)

**Review by tommyettinger — APPROVED** [review]
This looks very clean and seems like a simple fix. The test helps provide even more reassurance, thanks for that, and also thanks for using placeholder graphics to avoid any license concerns. I don't use Tiled myself, so I'm not especially qualified to review this, but no one else seems to be reviewing TMX PRs... I approve it, whatever that's worth.

**Comment by MobiDevelop** [comment]
The problem fixed here is in the dependency listing - the description of the PR doesn't make that particularly clear. I don't think that tileset tags can exist recursively, based on a review of the latest map spec, but I suppose it doesn't really hurt to let that be recursive. The important part here is for imagelayer elements to be read recursively during the dependency listing phase.

File: tests/gdx-tests-android/assets/data/maps/tiled-groups-imagelayer/Lonesome_Forest_Summer_Tileset.tsx.tsx
[the remark this answers is not in the document]
Comment (TMalygin) [reply tests/gdx-tests-android/assets/data/maps/tiled-groups-imagelayer/Lonesome_Forest_Summer_Tileset.tsx.tsx:1]: no, it is  a typo.
This PR was created more than 1 year ago, and now, I don't have time to change anything on it.

Feel free to update this code on your way.

**Pull request merged**